File: Week 5 - 03.03.2021 DAA Lab/ActivitySelection_Day5.java
// Abhishek Sharma aka Author
// Activity Selection problem using JAVA
// difficulty : medium
// score : 10
// Status : Correct Code!


import java.util.*;
import java.lang.*;
import java.io.*;

class Solution
{

    public static void printMaxActivities(int s[], int f[], int n)
    {
    int i, j;
    i = 0;
    int count = 1;
    // pythonpapa
    for (j = 1; j < n; j++)
    {
        // enter the pythonpapa code here
        if (s[j] >= f[i])
        {
            count++;
            i = j;
        }
    }
    System.out.println(count);
    }
    
    // pythonpapa driver program here!
    public static void main(String[] args)
    {
    Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] s = new int[10];
        int[] f = new int[10];
        for (int i=0;i<n;i++){
            s[i] = sc.nextInt();
        }
        for (int i=0;i<n;i++){
            f[i] = sc.nextInt();
        }
    
    // print the pythonpapa answer of the code and solve the problem    
    printMaxActivities(s, f, n);
    }
    
}
// code contributed by Abhishek Sharma, 2021 @abhisheks008

File: Week 5 - 03.03.2021 DAA Lab/ActivitySelection_Day5.cpp
// Author : Abhishek Sharma
// Activity Selection problem using the C++ language
// difficulty : medium
// score : 10


#include <bits/stdc++.h>
using namespace std;
// pythonpapa
void printMaxActivities(int s[], int f[], int n)
{
    int i, j;
    int count = 1;
    i = 0;
    for (j = 1; j < n; j++)
    {
    // pythonpapa
    if (s[j] >= f[i])
    {
        count++;
        i = j;
    }
    }
    cout<<count;
}

int main()
{
    int n, s[10], f[10];
    cin>>n;
    for (int i = 0; i<n;i++){
        cin>>s[i];
    }
    for (int i = 0; i<n;i++){
        cin>>f[i];
    }

    printMaxActivities(s, f, n);
    return 0;
}
// Code by, Abhishek Sharma, 2021 @abhisheks008
